service/static/js/index.js

```diff
@@ -258,4 +260,117 @@ $(function () {
     });
   });
 
+  // ****************************************
+  // Delete a Product
+  // ****************************************
+
+  $("#delete-product").click(function (event) {
+    event.preventDefault();
+    var product_id = $("#wishlist-product-id").val();
+
+    var ajax = $.ajax({
+      type: "DELETE",
+      url: `/wishlists/products/` + parseInt(product_id),
+      contentType: "application/json",
+      data: '',
+    })
+
+    ajax.done(function(res){
+      clear_form_data();
+      flash_message("Success");
+    });
+
+    ajax.fail(function(res){
+      flash_message("Product doesn't exist!");
+      clear_table();
+    });
+  });
+
+  // ****************************************
+  // Update a Product
+  // ****************************************
+
+  $("#update-product").click(function (event) {
+    event.preventDefault();
+    var name = $("#wishlist-product-name").val();
+    var id = $("#wishlist-product-id").val();
+    var i_id = $("#wishlist-inventory-product-id").val();
+    var price = $("#wishlist-product-price").val();
+    var status = $("#wishlist-product-status").val();
+    var pic = $("#wishlist-product-pic").val();
+    var desc = $("#wishlist-product-description").val();
+
+    var data = {};
+
+    if(name){
```

Comment:
I think you should do this:
```const obj = { 
  requiredKey1: 1, 
  requiredKey2: 2,
  ...(flag1 && { optionalKey1: 5 }),
  ...(flag2 && { optionalKey2: 6, optionalKey3: 7 }),
  ...(flag1 && { optionalKey4: 8, optionalKey5: 9 })
};
```
Courtesy of [stackoverflow](https://stackoverflow.com/questions/47892127/succinct-concise-syntax-for-optional-object-keys-in-es6-es7). I think it looks cleaner this way.